.travis.yml

```diff
@@ -45,6 +63,16 @@ matrix:
   #          CONDA_DEPENDENCIES='sphinx cython numpy pytest-cov'
   #          EVENT_TYPE='push pull_request cron'
 
+before_install:
+
+      # Test OSX without OpenMP support
+      # Since the matrix OSX tests use the OS shipped version of clang, they double up
+      # as exploritory tests for when the shipped version has automatic OpenMP support.
+      # These tests will then fail and at such a time a new one should be added
+      # to explicitly remove OpenMP support.
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export TEST_OPENMP="False";
+      else export TEST_OPENMP="True"; fi
```

Comment:
Minor: I'd set TEST_OPENMP=True as a global higher up and just set it to False on MacOS X

Reply:
It's never good design to split conditionals like this. State logic should be as independent as it can be and contained within a single place. Splitting it out increases complexity and dependency, this is especially true here when considering stage overloading. Basically, it should only be set once and since this can't exist as a global (due to it be conditional on OS), it needs to exist within an actual eval stage.

If `before_install` is overloaded and `TEST_OPENMP` not set, the test will raise an exception [here](https://github.com/astropy/astropy-helpers/pull/382/files#diff-58c5d1bcd09b08d0f8c8144ada928f7eR22) alerting the developer that they have forgotten to add this to their overloaded version.